src/Sarif/Readers/ResultDiffingVisitor.cs

```diff
@@ -31,7 +31,7 @@ public bool Diff(IEnumerable<Result> actual)
 
             this.AbsentResults = this.SharedResults;
 
-            this.SharedResults = new HashSet<Result>();
+            this.SharedResults = new HashSet<Result>(ResultEqualityComparer.Instance);
```

Comment:
This works, but that property is actually `internal`. The public API is `Result.ValueComparer`.